src/aiokatcp/sensor.py

```diff
@@ -236,19 +254,17 @@ def params(self) -> List[bytes]:
         else:
             return []
 
-    def attach(self, observer: Observer) -> None:
+    def attach(self, observer: Observer[_T]) -> None:
         sig = inspect.signature(observer)
+        print(sig)
         if "old_reading" in sig.parameters:
             self._change_aware_observers.add(observer)  # type: ignore
         else:
             self._classic_observers.add(observer)  # type: ignore
 
-    def detach(self, observer: Observer) -> None:
-        sig = inspect.signature(observer)
-        if "old_reading" in sig.parameters:
-            self._change_aware_observers.discard(observer)  # type: ignore
-        else:
-            self._classic_observers.discard(observer)  # type: ignore
+    def detach(self, observer: Observer[_T]) -> None:
+        self._change_aware_observers.discard(observer)  # type: ignore
+        self._classic_observers.discard(observer)  # type: ignore
```

Comment:
Maybe a comment to justify ignoring the type e.g.
```suggestion
        # It's simpler to just discard from both sets (and ignore the type) than to do inspection.
        self._change_aware_observers.discard(observer)  # type: ignore
        self._classic_observers.discard(observer)  # type: ignore
```